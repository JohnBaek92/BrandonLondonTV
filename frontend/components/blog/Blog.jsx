import React from "react";
import UnderConstruction from "../misc/UnderConstruction";
import {Link} from "react-router-dom";

class Blog extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount() {
    fetch("/blogs")
      .then(res => {
        return res.json();
      })
      .then(blogs => {
        this.setState({ blogs });
      })
      .catch(err => this.setState({ error: err.message }));

  }

  render() {
    const { blogs } = this.state;
    if (!blogs) return null;
    let posts = blogs.map(foo => <BlogSnippet blog={foo} />);
    return <div className="makeFlex">{posts}</div>;
  }
}

export default Blog;

const makeDate = (date) => {
  return new Date(Date.parse(date)).toLocaleDateString()
}

const BlogSnippet = ({ blog }) => (
  <div className="blog snippet card">
    <div className="card-image">
      <figure className="image is-3by2">
        <img src={blog.image} alt="" className="image" />
      </figure>
    </div>
    <div className="card-content">
      <div className="media">
        <div className="media-content">
          <p
            className="title is-2 has-text-centered"
            style={{ color: "rgb(3, 167, 151)", paddingBottom: "0.2em" }}
          >
            {blog.title}
          </p>
          <p
            className="subtitle is-4 has-text-centered"
            style={{ color: "rgb(27, 0, 255)" }}
          >
            {blog.subtitle}
          </p>
        </div>
      </div>
    </div>
    <div className="content">
      {blog.blurb}
      <br />
      <time dateTime="2016-1-1">Published: {makeDate(blog.date)}</time>
      <br />
      <br/>
      <div
        className="notification hoverReadMore"
      >
        <Link to={`/blog/` + blog.id}>Read More</Link>
      </div>
    </div>
  </div>
);
