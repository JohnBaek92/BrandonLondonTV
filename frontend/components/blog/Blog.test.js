import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Blog from "./Blog";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Blog component should render as expected", () => {
  const component = shallow(<Blog />);
  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
});
