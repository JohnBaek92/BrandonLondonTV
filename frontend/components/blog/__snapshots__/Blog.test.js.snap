// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component should render as expected 1`] = `""`;
